lib/new_relic/agent/utilization_data.rb

```diff
@@ -84,10 +86,32 @@ def to_collector_hash
         result
       end
 
+      def append_ecs_info(collector_hash)
+        return unless Agent.config[:'utilization.detect_aws']
```

Comment:
I'm not sure I understand why we want to return early here. Is it because ECS is a service of AWS, so if someone specifies they're running in AWS, we won't get more specific?

Reply:
If someone has detect_aws turned off, we want to skip the ecs check as well since they are both controlled by the aws config.

Reply:
This is all new to me! So basically because Fargate is an AWS service, it falls in the category of 'detect aws'?

Reply:
Nevermind - I see this is all explained in the spec!

Reply:
🤦 - I should've read the spec first. Thank you!